### DISCUSSION TITLE
Add interpretive pages as data objects?

### DISCUSSION THREAD

**Discussion by kyledenlinger** [CollectionBuilder/collectionbuilder.github.io#91] [Q&A]
I'm still playing around with CB but I think it provides one of the more student-friendly experiences for collaborating on a digital collection. Thanks so much for your work on this.

I'm working with a faculty member whose students will be curating objects (photos, etc.) and assigning them metadata. They'll then be writing short entries, including citations, that integrate these objects. I briefly considered hijacking an object's description field, but that's just too limiting.

What might I need to do to accomplish the following?
- Assign pages metadata, especially location, so they appear on maps
- Have interpretive pages appear as cards when browsing by subjects
- Have backlinks from objects to any pages that integrate them

It's very possible I'm approaching this in the wrong way. Any advice is appreciated. Thanks!

**Comment by dcnb** [comment]
Hi @kyledenlinger! 

Thanks for your kind words about CB! I'm glad you are enjoying it, and this seems like a really great class project. 

So I think I would approach your issue in a sort of hybrid way. It will require some customization, and so I would recommend you use [CollectionBuilder-CSV](https://github.com/CollectionBuilder/collectionbuilder-csv) -- which might be a little challenging, but it can essentially be used/worked with in the same way as GH via the GitHub web interface.  (Let us know if you need any instruction on that -- @juliastone0729 is available this summer to help with that type of training!)

My suggestion would then be to follow your instinct and add the short entries as objects -- this way they'd show up in the map and browse, etc. -- you can also give them a featured image via the `image_small` and `image_thumb` fields.  I'd then give them a special display_template in their metadata called "interpretive" or something like that. After that, you'll need to create a folder in your _includes folder called "interpretive" where the students can write their short pieces in markdown (or you can put them there). This way there's more leeway for writing but the pieces are still included as part of the collection itself. 

I checked to see if that would work. Here's a rough sketch of what that might look like: 

https://github.com/dcnb/test-interpretiveobject

You might look at the [commits](https://github.com/dcnb/test-interpretiveobject/commits/main) to see what I did. 

As for adding backlinks from objects, you could do that via the metadata as well. My former colleague @mshannon95 did that for our Harvester special collections blog. Here's the line where that starts:  

https://github.com/uidaholib/spec-lumber/blob/84cbb023c69191601d81ae71688183c8fc406879/_layouts/item.html#L75

So for items that match blog posts, SPEC staff just add the title of the blog post and the link in separate fields (and if there are more than one they separate by semi-colon).  You would just add similar code to the file at `_layouts/item/item-page-base.html`

You can see how that shows up here: https://harvester.lib.uidaho.edu/collection/items/lumber1424.html
(sorry for the moldy image!)

So this is a definitely an advanced customization, but hopefully that gives you some sense of where to go. Please let us know if we're missing something, or if you'd like to jump on a quick zoom to figure some of this out. Happy to help!

Very cool project! Good luck! and let us know how it's going,
@dcnb

**Reply by evanwill** [reply]
@kyledenlinger a few other approach ideas I might add-- 
to render markdown from the metadata csv, we can add in "markdownify" to where the value appears on the page layout so that any markdown included in the field would be rendered correctly. 

Another approach, if the focus is on interpretive writing (rather than the csv metadata), you can actually do an entire collection based on folder of markdown files rather than a spreadsheet. The markdown files would be in a folder "_items", each file is the equivalent of one row of your spreadsheet. In the yaml front matter, add structured metadata fields which act exactly the same as the columns of a metadata spreadsheet, which can be used to populate Browse, Map, etc etc. In the body of the markdown file, you write your interpretative content, which would allow more in depth writing (than trying to do it in a spreadsheet), and would be displayed on the item page. I have implemented this before if you are interested!
